app/javascript/components/cloud-volume-form/attach-detach-cloud-volume-form.jsx

```diff
@@ -19,10 +19,20 @@ const AttachDetachCloudVolumeForm = ({ recordId, isAttach, vmChoices }) => {
     }));
   };
 
+  const dropdownOptions = [];
+  dropdownChoices.forEach((opt) => {
+    dropdownOptions.push({ label: opt[0], value: opt[1].toString() });
+  });
+
   useEffect(() => {
-    if (isLoading && isAttach) {
+    if (isLoading && isAttach && dropdownLabel == "Instance") {
       API.options(`/api/cloud_volumes/${recordId}?option_action=attach`)
         .then(loadSchema());
+    } else if (isLoading && isAttach && dropdownLabel == "Volume") {
+      // NOTE We assume all Volumes passed in are of the same provider type as the chosen Instance
+      const refVolumeType = dropdownOptions[0]['value'];
```

Comment:
@Fryguy or @agrare can y'all confirm for me that it's not possible to attach a Volume to an Instance of a different type? (ie attach an Amazon Volume to an Openstack Instance)

Reply:
That is definitely not possible